src/lib/types.ts

```diff
@@ -105,3 +109,10 @@ export type OAuth2EndpointsInput = Partial<OAuth2Endpoints>;
 export interface OAuth2Options {
   endpoints?: OAuth2EndpointsInput;
 }
+
+export type PKCEAlgorithm = (typeof supportedPkceAlgorithms)[number];
```

Comment:
I feel that this should be more of an `enum`. Maybe it's a syntax I'm not used to, and it actually behaves as an enum?

Reply:
Correct, it uses a value of a constant and creates an enum out of it!